### PR TITLE
Add docs directory and placeholder file handling doc

### PR DESCRIPTION
**Pull request by aweiteka**

### CODE REVIEW

**Comment by aweiteka** [comment]
A start to addressing this: https://github.com/projectatomic/atomicapp/issues/223

File: docs/file_handling.asciidoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Installing an Atomic App means to download the artifacts and sample answerfile.
+
+== Developing and Debugging
+
+Image developers may run the root container and point to a Nulecule directory on the local system.
+
+== Directories
+
+* `/tmp/<atomicapp_name>`: Host directory for temporary Nulecule files. May be overriden.
+* `/tmp/<atomicapp_name>/.workdir`: Host directory for "rendered" artifact files with variable substitution.
```
Comment (vpavlin) [inline docs/file_handling.asciidoc]: Maybe replace "rendered" with templated? (If that's a real word:D)

Comment (charliedrage) [inline docs/file_handling.asciidoc]: Or
```
Host directory for artifact template files with variable substitution
```

**Comment by goern** [comment]
@aweiteka could you have a look at it again, i think @charliedrage added a docs/providers.md and you move it in this PR from ./ to docs/ ... it should break, shouldnt it?

**Comment by aweiteka** [comment]
@goern please review

**Comment by charliedrage** [comment]
@aweiteka could we have this in markdown so it's consistent to the other docs?

**Comment by aweiteka** [comment]
updated

**Comment by charliedrage** [comment]
LGTM :+1:

**Pull request merged**